InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -0,0 +1,85 @@
+/* global instantsearch */
+
+function getCategorySlug(name) {
+  return name.replace(/\s/g, '-');
+}
+
+function getCategoryName(slug) {
+  return slug.replace(/-/g, ' ');
+}
+
+const router = instantsearch.routers.history({
+  windowTitle({ category, q }) {
+    const query = q ? `Results for "${q}"` : 'Search';
+
+    if (category) {
+      return [category, query].join(' – ');
+    }
+
+    return query;
+  },
+  createURL({ qsModule, routeState, location }) {
+    const { protocol, hostname, port = '', hash } = location;
+    const portWithPrefix = port === '' ? '' : `:${port}`;
+    const pathname = routeState.category
+      ? `/${getCategorySlug(routeState.category)}/`
+      : '';
+
+    const searchState = {};
+    if (routeState.q) searchState.q = encodeURIComponent(routeState.q);
+    if (routeState.p !== 1) searchState.p = routeState.p;
+    if (routeState.brands) searchState.brands = routeState.brands;
+
+    const queryString = qsModule.stringify(searchState, {
+      addQueryPrefix: true,
+    });
+
+    return `${protocol}//${hostname}${portWithPrefix}${pathname}${queryString}${hash}`;
+  },
+  parseURL({ qsModule, location }) {
+    const routeStateString = location.pathname.split('/')[1];
+    const category = getCategoryName(routeStateString.split('/')[0]);
+    const queryParameters = qsModule.parse(location.search.slice(1));
+    const { q = '', p, brands } = queryParameters;
```

Comment:
Is there a reason why we don’t assign on those values on the line above?

Reply:
Mainly for clarity for our users.